### ISSUE TITLE
Unknown error when access a remote share after remote sharing has been disabled

### ISSUE THREAD

**Issue by jnfrmarks** [owncloud/core#13578]
### Steps to reproduce

1. Install owncloud on two systems
2. on system one, remote share a file with a user on system two
3. on system two, accept share and validate access
4. on system one, as the admin user uncheck the "Allow users on this server to send shares to other servers" checkbox
5. on system two, access the share

### Expected behaviour

Unknown:  Should the remote user still have access to the share?

### Actual behaviour

An "unknown error" is thrown.

![p253](https://cloud.githubusercontent.com/assets/8480343/5845734/8bf93e6e-a18c-11e4-9d58-ab8e82803056.PNG)

### Server log file

```

{"reqId":"a0d1a685efd816381e0d1a1f61a4bfee","remoteAddr":"192.168.1.218","app":"files","message":"Exception: {\"Message\":\"The passed data is not valid XML\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/html\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/Sabre\\\/DAV\\\/Client.php(181): Sabre\\\\DAV\\\\Client->parseMultiStatus(false)\\n#1 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/files\\\/storage\\\/dav.php(513): Sabre\\\\DAV\\\\Client->propFind('', Array)\\n#2 \\\/var\\\/www\\\/html\\\/owncloud\\\/apps\\\/files_sharing\\\/lib\\\/external\\\/storage.php(144): OC\\\\Files\\\\Storage\\\\DAV->hasUpdated('', 1421875827)\\n#3 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/files\\\/storage\\\/wrapper\\\/wrapper.php(357): OCA\\\\Files_Sharing\\\\External\\\\Storage->hasUpdated('', 1421875827)\\n#4 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/files\\\/storage\\\/wrapper\\\/wrapper.php(357): OC\\\\Files\\\\Storage\\\\Wrapper\\\\Wrapper->hasUpdated('', 1421875827)\\n#5 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/files\\\/cache\\\/watcher.php(67): OC\\\\Files\\\\Storage\\\\Wrapper\\\\Wrapper->hasUpdated('', 1421875827)\\n#6 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/files\\\/view.php(932): OC\\\\Files\\\\Cache\\\\Watcher->checkUpdate('', Array)\\n#7 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/files\\\/filesystem.php(782): OC\\\\Files\\\\View->getFileInfo('\\\/picdir', true)\\n#8 \\\/var\\\/www\\\/html\\\/owncloud\\\/apps\\\/files\\\/ajax\\\/list.php(12): OC\\\\Files\\\\Filesystem::getFileInfo('\\\/picdir')\\n#9 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/route\\\/route.php(135) : runtime-created function(1): require_once('\\\/var\\\/www\\\/html\\\/o...')\\n#10 [internal function]: __lambda_func(Array)\\n#11 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/route\\\/router.php(250): call_user_func('?lambda_1697', Array)\\n#12 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/base.php(763): OC\\\\Route\\\\Router->match('\\\/apps\\\/files\\\/aja...')\\n#13 \\\/var\\\/www\\\/html\\\/owncloud\\\/index.php(28): OC::handleRequest()\\n#14 {main}\",\"File\":\"\\\/var\\\/www\\\/html\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/Sabre\\\/DAV\\\/Client.php\",\"Line\":540}","level":4,"time":"2015-01-21T21:37:10+00:00"}


```

### Server Host info:

Ubuntu 14.04
PHP 5.5.9

### Server configuration

ownCloud Enterprise Edition 8.0 beta 1 (daily) Build:2015-01-21T04:13:37+00:00

**Comment by karlitschek** [comment]
@schiesbn

**Comment by MorrisJobke** [comment]
@PVince81 Is this maybe fixed by #13525

**Comment by PVince81** [comment]
I haven't tested this specific case, but it might be fixed by that one, yes.

**Comment by MorrisJobke** [comment]
@PVince81 Let me check that :) Then I will add the "fixes #..." statement in this PR

**Comment by DeepDiver1975** [comment]
@MorrisJobke did you test this?
@jnfrmarks Can you please retest this cas with OC8 beta 2? THX

**Comment by MorrisJobke** [comment]
I had problems, that are fixed now. Let me look at this later

**Comment by MorrisJobke** [comment]
Nothing in the logs of server one.

Log server two: 

```
{"reqId":"30fb20f13ed063cb6f3d26d91897654c","remoteAddr":"127.0.0.1","app":"files","message":"Exception: {\"Message\":\"The passed data is not valid XML\",\"Code\":0,\"Trace\":\"#0 \\\/home\\\/mjob\\\/Projekte\\\/owncloud\\\/master\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/Sabre\\\/DAV\\\/Client.php(181): Sabre\\\\DAV\\\\Client->parseMultiStatus(false)\\n#1 \\\/home\\\/mjob\\\/Projekte\\\/owncloud\\\/master\\\/lib\\\/private\\\/files\\\/storage\\\/dav.php(532): Sabre\\\\DAV\\\\Client->propFind('', Array)\\n#2 \\\/home\\\/mjob\\\/Projekte\\\/owncloud\\\/master\\\/apps\\\/files_sharing\\\/lib\\\/external\\\/storage.php(144): OC\\\\Files\\\\Storage\\\\DAV->hasUpdated('', 1422870413)\\n#3 \\\/home\\\/mjob\\\/Projekte\\\/owncloud\\\/master\\\/lib\\\/private\\\/files\\\/storage\\\/wrapper\\\/wrapper.php(357): OCA\\\\Files_Sharing\\\\External\\\\Storage->hasUpdated('', 1422870413)\\n#4 \\\/home\\\/mjob\\\/Projekte\\\/owncloud\\\/master\\\/lib\\\/private\\\/files\\\/cache\\\/watcher.php(67): OC\\\\Files\\\\Storage\\\\Wrapper\\\\Wrapper->hasUpdated('', 1422870413)\\n#5 \\\/home\\\/mjob\\\/Projekte\\\/owncloud\\\/master\\\/lib\\\/private\\\/files\\\/view.php(941): OC\\\\Files\\\\Cache\\\\Watcher->checkUpdate('', Array)\\n#6 \\\/home\\\/mjob\\\/Projekte\\\/owncloud\\\/master\\\/lib\\\/private\\\/files\\\/filesystem.php(791): OC\\\\Files\\\\View->getFileInfo('\\\/asdasd', true)\\n#7 \\\/home\\\/mjob\\\/Projekte\\\/owncloud\\\/master\\\/apps\\\/files\\\/ajax\\\/list.php(12): OC\\\\Files\\\\Filesystem::getFileInfo('\\\/asdasd')\\n#8 \\\/home\\\/mjob\\\/Projekte\\\/owncloud\\\/master\\\/lib\\\/private\\\/route\\\/route.php(135) : runtime-created function(1): require_once('\\\/home\\\/mjob\\\/Proj...')\\n#9 [internal function]: __lambda_func(Array)\\n#10 \\\/home\\\/mjob\\\/Projekte\\\/owncloud\\\/master\\\/lib\\\/private\\\/route\\\/router.php(250): call_user_func('\\\\x00lambda_1127', Array)\\n#11 \\\/home\\\/mjob\\\/Projekte\\\/owncloud\\\/master\\\/lib\\\/base.php(763): OC\\\\Route\\\\Router->match('\\\/apps\\\/files\\\/aja...')\\n#12 \\\/home\\\/mjob\\\/Projekte\\\/owncloud\\\/master\\\/index.php(28): OC::handleRequest()\\n#13 {main}\",\"File\":\"\\\/home\\\/mjob\\\/Projekte\\\/owncloud\\\/master\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/Sabre\\\/DAV\\\/Client.php\",\"Line\":542}","level":4,"time":"2015-02-02T09:47:18+00:00","method":"GET","url":"\/master\/index.php\/apps\/files\/ajax\/list.php?dir=%2Fasdasd&sort=name&sortdirection=asc"}
```

I still see the yellow notification: "Unknown error". (also the response just contains the "Unknown error")

**Comment by MorrisJobke** [comment]
Prettyfied:

```
{
	"reqId":"30fb20f13ed063cb6f3d26d91897654c",
	"remoteAddr":"127.0.0.1",
	"app":"files",
	"message":"Exception: {
		"Message":"The passed data is not valid XML",
		"Code":0,
		"Trace":"
			#0 3rdparty/sabre/dav/lib/Sabre/DAV/Client.php(181): Sabre\\DAV\\Client->parseMultiStatus(false)
			#1 lib/private/files/storage/dav.php(532): Sabre\\DAV\\Client->propFind('', Array)
			#2 apps/files_sharing/lib/external/storage.php(144): OC\\Files\\Storage\\DAV->hasUpdated('', 1422870413)
			#3 lib/private/files/storage/wrapper/wrapper.php(357): OCA\\Files_Sharing\\External\\Storage->hasUpdated('', 1422870413)
			#4 lib/private/files/cache/watcher.php(67): OC\\Files\\Storage\\Wrapper\\Wrapper->hasUpdated('', 1422870413)
			#5 lib/private/files/view.php(941): OC\\Files\\Cache\\Watcher->checkUpdate('', Array)
			#6 lib/private/files/filesystem.php(791): OC\\Files\\View->getFileInfo('/asdasd', true)
			#7 apps/files/ajax/list.php(12): OC\\Files\\Filesystem::getFileInfo('/asdasd')
			#8 lib/private/route/route.php(135) : runtime-created function(1): require_once('/home/mjob/Proj...')
			#9 [internal function]: __lambda_func(Array)
			#10 lib/private/route/router.php(250): call_user_func('\\x00lambda_1127', Array)
			#11 lib/base.php(763): OC\\Route\\Router->match('/apps/files/aja...')
			#12 index.php(28): OC::handleRequest()
			#13 {main}",
		"File":"3rdparty/sabre/dav/lib/Sabre/DAV/Client.php",
		"Line":542
	}",
	"level":4,
	"time":"2015-02-02T09:47:18+00:00",
	"method":"GET",
	"url":"/index.php/apps/files/ajax/list.php?dir=%2Fasdasd&sort=name&sortdirection=asc"
}
```

**Comment by DeepDiver1975** [comment]
> "Message":"The passed data is not valid XML",

this means the server on the other side if not returning proper xml - most probably is ran into some exception which is rendered as html.

@MorrisJobke Any exception on the other side? Maybe you can please a breakpoint inside parseMultiStatus() and see the content? THX

**Comment by jnfrmarks** [comment]
@DeepDiver1975 @MorrisJobke 

I retested with 8.1 alpha 3. There is still a problem.  On the remote system, the UI appears hung or non responsive until I refreshed it.  This is in the remote system server log:

```

{"reqId":"ytl+62Vl8x3CnsCbDtQc","remoteAddr":"","app":"PHP","message":"touch(): Unable to create file \/var\/www\/html\/owncloud\/data\/owncloud.log because No such file or directory at \/var\/www\/html\/owncloud\/lib\/private\/log\/owncloud.php#60","level":3,"time":"2015-05-05T14:14:59+00:00"}
{"reqId":"K67orjcCLh7l7ueF2TRR","remoteAddr":"x.x.x.x","app":"user_ldap","message":"Could not autodetect the UUID attribute","level":3,"time":"2015-05-05T15:37:40+00:00"}
{"reqId":"DXdhjyfyGMR1Cw5gyNIb","remoteAddr":"x.x.x.x","app":"PHP","message":"fopen(\/var\/www\/html\/owncloud\/data\/abc\/thumbnails\/12\/2048-2048-max.png): failed to open stream: No such file or directory at \/var\/www\/html\/owncloud\/lib\/private\/files\/storage\/local.php#256","level":3,"time":"2015-05-05T15:37:47+00:00"}
{"reqId":"DXdhjyfyGMR1Cw5gyNIb","remoteAddr":"x.x.x.x","app":"PHP","message":"fopen(\/): failed to open stream: Is a directory at \/var\/www\/html\/owncloud\/apps\/files_locking\/lib\/lock.php#135","level":3,"time":"2015-05-05T15:37:47+00:00"}
{"reqId":"DXdhjyfyGMR1Cw5gyNIb","remoteAddr":"x.x.x.x","app":"index","message":"Exception: {\"Exception\":\"OCP\\\\Files\\\\LockNotAcquiredException\",\"Message\":\"Could not obtain lock type 2 on \\\"\\\/\\\".\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/html\\\/owncloud\\\/apps\\\/files_locking\\\/lib\\\/lockingwrapper.php(44): OCA\\\\Files_Locking\\\\Lock->addLock(2, NULL)\\n#1 \\\/var\\\/www\\\/html\\\/owncloud\\\/apps\\\/files_locking\\\/lib\\\/lockingwrapper.php(93): OCA\\\\Files_Locking\\\\LockingWrapper->getLock('thumbnails\\\/12\\\/2...', 2)\\n#2 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/files\\\/view.php(936): OCA\\\\Files_Locking\\\\LockingWrapper->file_put_contents('thumbnails\\\/12\\\/2...', '\\\\x89PNG\\\\r\\\\n\\\\x1A\\\\n\\\\x00\\\\x00\\\\x00\\\\rIHD...')\\n#3 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/files\\\/view.php(554): OC\\\\Files\\\\View->basicOperation('file_put_conten...', 'thumbnails\\\/12\\\/2...', Array, '\\\\x89PNG\\\\r\\\\n\\\\x1A\\\\n\\\\x00\\\\x00\\\\x00\\\\rIHD...')\\n#4 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/preview.php(856): OC\\\\Files\\\\View->file_put_contents('thumbnails\\\/12\\\/2...', '\\\\x89PNG\\\\r\\\\n\\\\x1A\\\\n\\\\x00\\\\x00\\\\x00\\\\rIHD...')\\n#5 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/preview.php(816): OC\\\\Preview->storeMaxPreview('thumbnails\\\/12\\\/')\\n#6 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/preview.php(521): OC\\\\Preview->generatePreview(12)\\n#7 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/preview.php(544): OC\\\\Preview->getPreview()\\n#8 \\\/var\\\/www\\\/html\\\/owncloud\\\/core\\\/ajax\\\/preview.php(61): OC\\\\Preview->showPreview()\\n#9 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/route\\\/route.php(153) : runtime-created function(1): require_once('\\\/var\\\/www\\\/html\\\/o...')\\n#10 [internal function]: __lambda_func(Array)\\n#11 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/route\\\/router.php(272): call_user_func('\\\\x00lambda_384', Array)\\n#12 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/base.php(853): OC\\\\Route\\\\Router->match('\\\/core\\\/preview.p...')\\n#13 \\\/var\\\/www\\\/html\\\/owncloud\\\/index.php(40): OC::handleRequest()\\n#14 {main}\",\"File\":\"\\\/var\\\/www\\\/html\\\/owncloud\\\/apps\\\/files_locking\\\/lib\\\/lock.php\",\"Line\":251}","level":4,"time":"2015-05-05T15:37:47+00:00"}
{"reqId":"WTV3uF5andMtg9nZAgfU","remoteAddr":"x.x.x.x","app":"PHP","message":"fopen(\/var\/www\/html\/owncloud\/data\/abc\/thumbnails\/12\/2048-2048-max.png): failed to open stream: No such file or directory at \/var\/www\/html\/owncloud\/lib\/private\/files\/storage\/local.php#256","level":3,"time":"2015-05-05T15:37:51+00:00"}
{"reqId":"WTV3uF5andMtg9nZAgfU","remoteAddr":"x.x.x.x","app":"PHP","message":"fopen(\/): failed to open stream: Is a directory at \/var\/www\/html\/owncloud\/apps\/files_locking\/lib\/lock.php#135","level":3,"time":"2015-05-05T15:37:51+00:00"}
{"reqId":"WTV3uF5andMtg9nZAgfU","remoteAddr":"x.x.x.x","app":"index","message":"Exception: {\"Exception\":\"OCP\\\\Files\\\\LockNotAcquiredException\",\"Message\":\"Could not obtain lock type 2 on \\\"\\\/\\\".\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/html\\\/owncloud\\\/apps\\\/files_locking\\\/lib\\\/lockingwrapper.php(44): OCA\\\\Files_Locking\\\\Lock->addLock(2, NULL)\\n#1 \\\/var\\\/www\\\/html\\\/owncloud\\\/apps\\\/files_locking\\\/lib\\\/lockingwrapper.php(93): OCA\\\\Files_Locking\\\\LockingWrapper->getLock('thumbnails\\\/12\\\/2...', 2)\\n#2 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/files\\\/view.php(936): OCA\\\\Files_Locking\\\\LockingWrapper->file_put_contents('thumbnails\\\/12\\\/2...', '\\\\x89PNG\\\\r\\\\n\\\\x1A\\\\n\\\\x00\\\\x00\\\\x00\\\\rIHD...')\\n#3 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/files\\\/view.php(554): OC\\\\Files\\\\View->basicOperation('file_put_conten...', 'thumbnails\\\/12\\\/2...', Array, '\\\\x89PNG\\\\r\\\\n\\\\x1A\\\\n\\\\x00\\\\x00\\\\x00\\\\rIHD...')\\n#4 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/preview.php(856): OC\\\\Files\\\\View->file_put_contents('thumbnails\\\/12\\\/2...', '\\\\x89PNG\\\\r\\\\n\\\\x1A\\\\n\\\\x00\\\\x00\\\\x00\\\\rIHD...')\\n#5 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/preview.php(816): OC\\\\Preview->storeMaxPreview('thumbnails\\\/12\\\/')\\n#6 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/preview.php(521): OC\\\\Preview->generatePreview(12)\\n#7 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/preview.php(544): OC\\\\Preview->getPreview()\\n#8 \\\/var\\\/www\\\/html\\\/owncloud\\\/core\\\/ajax\\\/preview.php(61): OC\\\\Preview->showPreview()\\n#9 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/route\\\/route.php(153) : runtime-created function(1): require_once('\\\/var\\\/www\\\/html\\\/o...')\\n#10 [internal function]: __lambda_func(Array)\\n#11 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/route\\\/router.php(272): call_user_func('\\\\x00lambda_129', Array)\\n#12 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/base.php(853): OC\\\\Route\\\\Router->match('\\\/core\\\/preview.p...')\\n#13 \\\/var\\\/www\\\/html\\\/owncloud\\\/index.php(40): OC::handleRequest()\\n#14 {main}\",\"File\":\"\\\/var\\\/www\\\/html\\\/owncloud\\\/apps\\\/files_locking\\\/lib\\\/lock.php\",\"Line\":251}","level":4,"time":"2015-05-05T15:37:51+00:00"}
{"reqId":"AdwESX2j5N846O+bkpAb","remoteAddr":"x.x.x.x","app":"index","message":"Exception: {\"Exception\":\"InvalidArgumentException\",\"Message\":\"The body passed to parseMultiStatus could not be parsed. Is it really xml?\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/html\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Client.php(239): Sabre\\\\DAV\\\\Client->parseMultiStatus('')\\n#1 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/files\\\/storage\\\/dav.php(681): Sabre\\\\DAV\\\\Client->propFind('', Array)\\n#2 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/files\\\/storage\\\/dav.php(659): OC\\\\Files\\\\Storage\\\\DAV->getPermissions('')\\n#3 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/files\\\/storage\\\/wrapper\\\/wrapper.php(169): OC\\\\Files\\\\Storage\\\\DAV->isUpdatable('')\\n#4 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/files\\\/storage\\\/wrapper\\\/wrapper.php(169): OC\\\\Files\\\\Storage\\\\Wrapper\\\\Wrapper->isUpdatable('')\\n#5 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/files\\\/view.php(938): OC\\\\Files\\\\Storage\\\\Wrapper\\\\Wrapper->isUpdatable('')\\n#6 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/files\\\/view.php(391): OC\\\\Files\\\\View->basicOperation('isUpdatable', '\\\/\\\/pre_128.txt')\\n#7 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/files\\\/filesystem.php(642): OC\\\\Files\\\\View->isUpdatable('\\\/\\\/pre_128.txt')\\n#8 \\\/var\\\/www\\\/html\\\/owncloud\\\/apps\\\/files_texteditor\\\/ajax\\\/loadfile.php(36): OC\\\\Files\\\\Filesystem::isUpdatable('\\\/\\\/pre_128.txt')\\n#9 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/route\\\/route.php(153) : runtime-created function(1): require_once('\\\/var\\\/www\\\/html\\\/o...')\\n#10 [internal function]: __lambda_func(Array)\\n#11 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/route\\\/router.php(272): call_user_func('\\\\x00lambda_283', Array)\\n#12 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/base.php(853): OC\\\\Route\\\\Router->match('\\\/apps\\\/files_tex...')\\n#13 \\\/var\\\/www\\\/html\\\/owncloud\\\/index.php(40): OC::handleRequest()\\n#14 {main}\",\"File\":\"\\\/var\\\/www\\\/html\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Client.php\",\"Line\":454}","level":4,"time":"2015-05-05T15:38:30+00:00"}

```

**Comment by jnfrmarks** [comment]
I filed https://github.com/owncloud/files_locking/issues/23 for the thumbnails issue in the server log

**Comment by jnfrmarks** [comment]
This is the error in the remote system thrown when trying to access the file after remote sharing was disabled on the first system:

```

{"reqId":"3TYbFEijSBT5\/pMT9HUa","remoteAddr":"x.x.x.x","app":"index","message":"Exception: {\"Exception\":\"InvalidArgumentException\",\"Message\":\"The body passed to parseMultiStatus could not be parsed. Is it really xml?\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/html\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Client.php(239): Sabre\\\\DAV\\\\Client->parseMultiStatus('')\\n#1 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/files\\\/storage\\\/dav.php(681): Sabre\\\\DAV\\\\Client->propFind('', Array)\\n#2 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/files\\\/storage\\\/dav.php(659): OC\\\\Files\\\\Storage\\\\DAV->getPermissions('')\\n#3 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/files\\\/storage\\\/wrapper\\\/wrapper.php(169): OC\\\\Files\\\\Storage\\\\DAV->isUpdatable('')\\n#4 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/files\\\/storage\\\/wrapper\\\/wrapper.php(169): OC\\\\Files\\\\Storage\\\\Wrapper\\\\Wrapper->isUpdatable('')\\n#5 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/files\\\/view.php(938): OC\\\\Files\\\\Storage\\\\Wrapper\\\\Wrapper->isUpdatable('')\\n#6 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/files\\\/view.php(391): OC\\\\Files\\\\View->basicOperation('isUpdatable', '\\\/\\\/pre_128.txt')\\n#7 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/files\\\/filesystem.php(642): OC\\\\Files\\\\View->isUpdatable('\\\/\\\/pre_128.txt')\\n#8 \\\/var\\\/www\\\/html\\\/owncloud\\\/apps\\\/files_texteditor\\\/ajax\\\/loadfile.php(36): OC\\\\Files\\\\Filesystem::isUpdatable('\\\/\\\/pre_128.txt')\\n#9 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/route\\\/route.php(153) : runtime-created function(1): require_once('\\\/var\\\/www\\\/html\\\/o...')\\n#10 [internal function]: __lambda_func(Array)\\n#11 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/private\\\/route\\\/router.php(272): call_user_func('\\\\x00lambda_451', Array)\\n#12 \\\/var\\\/www\\\/html\\\/owncloud\\\/lib\\\/base.php(853): OC\\\\Route\\\\Router->match('\\\/apps\\\/files_tex...')\\n#13 \\\/var\\\/www\\\/html\\\/owncloud\\\/index.php(40): OC::handleRequest()\\n#14 {main}\",\"File\":\"\\\/var\\\/www\\\/html\\\/owncloud\\\/3rdparty\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Client.php\",\"Line\":454}","level":4,"time":"2015-05-05T17:11:29+00:00"}

``

**Comment by PVince81** [comment]
Hmm, probably yet another specific exception / code path that we need to catch. The remote server doesn't return valid XML so the Sabre client is confused.

**Comment by PVince81** [comment]
I tried this on master (e1483f65c3cd800f90e962764b06bf2379ca4432) but I'm now getting "Storage not available", which is more acceptable than "Unknown error". I connected to a 8.0.3 server.

The problem is that the remote server is sending an HTML page in that case, for whatever reason.
I can try and fix "public.php/webdav" to return a proper error. But this means that the correct error will only be returned from newer servers. Connecting to older OC versions will still get the bad HTML page, and show "Storage not available", which I think is not too bad.
